### ISSUE TITLE
Extra spaces rendered after unicode alchemical symbols

### ISSUE THREAD

**Issue by sigurdo** [alacritty/alacritty#7239]
### System

OS: Windows 11 (in interaction with WSL2 instance)
Version: `alacritty 0.12.2 (9d9982d)`

### Logs

Font/Terminal size: `alacritty -vv`:

```
[0.000003900s] [INFO ] [alacritty] Welcome to Alacritty
[0.000572200s] [INFO ] [alacritty] Version 0.12.2 (9d9982d)
[0.005028900s] [INFO ] [alacritty] Configuration files loaded from:
                                     "C:\\Users\\sigur\\AppData\\Roaming\\alacritty\\alacritty.yml"
                                     "C:\\Users\\sigur\\AppData\\Roaming\\alacritty\\alacritty-theme\\themes\\monokai_charcoal.yaml"
[0.088886000s] [INFO ] [alacritty] Window scale factor: 1.5
[0.224636100s] [INFO ] [alacritty] Using WGL
[0.242199500s] [DEBUG] [alacritty] Loading "Fira Code" font
[0.246264500s] [INFO ] [alacritty] Running on NVIDIA GeForce GTX 1660 Ti/PCIe/SSE2
[0.256189300s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.283906400s] [DEBUG] [alacritty] Filling glyph cache with common glyphs
[0.286541000s] [INFO ] [alacritty] Cell size: 14 x 28
[0.286919300s] [INFO ] [alacritty] Padding: 0 x 0
[0.287327300s] [INFO ] [alacritty] Width: 2560, Height: 1440
[0.326420800s] [INFO ] [alacritty] PTY dimensions: 51 x 182
[0.336180500s] [INFO ] [alacritty] Initialisation complete
[0.341505200s] [DEBUG] [alacritty_terminal] Term::resize dimensions unchanged
```

### Issue

After a rendering a piece of text including unicode alchemical symbols, such as `🜎` (philosopher's stone), Alacritty adds one extra space per alchemical symbol after the first subsequent ANSI escape sequence. Demonstration: consider the following ANSI file [`demo.ansi`](https://sigurdt.no/alchemy-rendering/demo.ansi):

```
🜎.....(ESC)[0m.
🜎🜎🜎...(ESC)[0m.
🜎🜎....(ESC)[0m.🜎🜎....(ESC)[0m.🜎🜎....(ESC)[0m.
```

`(ESC)` represents an escape character, ``. Printing the file to the terminal produces the following result:

![image](https://github.com/alacritty/alacritty/assets/36281346/48d4bf12-2b3d-4e12-b017-ec10dee3b6ce)

According to the rust crate [`unicode_width`](https://docs.rs/unicode-width/latest/unicode_width/), all alchemical symbols should be half-width in both CJK and non-CJK contexts, even though many of them are quite wide, like e.g. `🜽` (orpiment), which is rendered full-width by VSCode, while `🜎` (philosopher's stone) is rendered half-width. Perfoming the above demonstration with `🜽` (orpiment) leads to the same behavior, but with a lot of overlapped rendering.

In other words, I think I see why proper rendering of alchemical symbols is challenging, but I can not see why the way it is handled by Alacritty should be correct.

For reference, in Windows Terminal, all alchemical symbols are rendered half-width, scaled down if necessary, and the results from the aforementioned demonstration looks like this:

![image](https://github.com/alacritty/alacritty/assets/36281346/1e1dd8be-daa5-4dfa-bc70-b57a4daae287)

**Comment by kchibisov** [comment]
That's how it looks for me. Sounds like ConPTY issue.

![picture](https://github.com/alacritty/alacritty/assets/27620401/39531d41-521c-4fb5-b4b5-576ad733a54d)

**Comment by sigurdo** [comment]
Oh, interesting... I see that there is described a config option `winpty_backend` in `alacritty_terminal/src/config/mod.rs` to use WinPTY backend even if ConPTY is available. Would you mind to quickly explain to me how I enable this?

**Comment by kchibisov** [comment]
there's no `winpty` anymore (and for a long time). The only option you have is to build alacritty from master, and use `conpty.dll`/`OpenConsole.exe` from other projects, putting them next to alacritty or in the `PATH`.

**Comment by sigurdo** [comment]
Thank you very much for the help! Now I finally managed to compile `conpty.dll` and `OpenConsole.exe` from the Windows Terminal source code, then I copied them to a folder in `PATH` and a self-compiled version of Alacritty will now have more proper behvior. Too large glyphs are not scaled down, like they are in the Windows Terminal, but at least this is a lot better.

For people facing the same issue, you can find more help and useful links in [this thread](https://github.com/microsoft/terminal/issues/8576).

**Comment by kchibisov** [comment]
Yeah, we don't scale down the glyphs, but the issue you were having is basically conpty broken width.

**Comment by lbergnehr** [comment]
I got this working correctly by putting conpty.dll and OpenConsole.exe next to alacritty.exe (release build -- did not build it myself). Got the binaries from this nuget package (simply a zip file): https://www.nuget.org/packages/CI.Microsoft.Windows.Console.ConPTY
